### ISSUE TITLE
Trigger rendering twice after upgrading to antd@4.85

### ISSUE THREAD

**Issue by felipepsq** [ant-design/ant-design#28006]
- [X] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/imwc3)

### Steps to reproduce
After reducing screen's size on breakpoint="lg", the sider trigger will appear, clicking on it will console.log the values twice - opening and closing the left menu. 

### What is expected?
Until version 4.8.4 of antd, this isn't happening, so it is triggering only once.

### What is actually happening?
Sider trigger is being triggered twice.

| Environment | Info |
|---|---|
| antd | 4.8.5 |
| React | 17.0.1 |
| System | Ubuntu 20.04.1 LTS |
| Browser |  Google Chrome 87.0.4280.66  |

---
As you can see, the code at codesandbox is the current antd example at https://ant.design/components/layout/.

**Comment by zombieJ** [comment]
Duplicate of #27975
